php/class-cache.php

```diff
@@ -0,0 +1,1019 @@
+<?php
+/**
+ * Cloudinary Logger, to collect logs and debug data.
+ *
+ * @package Cloudinary
+ */
+
+namespace Cloudinary;
+
+use Cloudinary\Cache\Cache_Point;
+use Cloudinary\Cache\File_System;
+use Cloudinary\Component\Setup;
+use Cloudinary\Settings\Setting;
+
+/**
+ * Plugin report class.
+ */
+class Cache extends Settings_Component implements Setup {
```

Comment:
@DavidCramer 

Given the length of this class, what do you think about separating the settings from the rest? It should help to reduce the size and they are isolated concepts. We have already a folder for cache, so we could have a `class-settings.php` inside it.